File: WFM-API/Models/EmployeeAppointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WFM_API.Models.Identity;


namespace WFM_API.Models
{
    public class EmployeeAppointment
    {
        [Required]
        public int Id { get; set; }


        [Required]
        public string? EmployeePID { get; set; }

        [ForeignKey("EmployeePID")]

        public AppUser? User { get; set; }

        [Required]
        public int TypeOfDayId { get; set; }

        [Required]
        public DateTime AppointMentDate { get; set; }

        [Required]
        public TimeSpan From { get; set; }
        [Required]
        public TimeSpan To { get; set; }
        public TypeOfDay? TypeOfDay { get; set; }

        public ICollection<EmpBreak>? Breaks { get; set; }
        public ICollection<EmpException>? Exceptions { get; set; }




    }
}

File: WFM-Angular/WFM-UI/src/app/Interfaces/EmployeeAppointment.ts
import { Time } from "@angular/common";
import { typeOfDay } from "./TypeOfDay";
import { Breaks } from "./Breaks";
import { Excption } from "./Excption";

export interface EmployeeAppointment{
        id:number,
        employeePID:string,
        typeOfDayId:number,
        appointMentDate:Date,
        from:Time,
        to:Time,
        typeOfDay:typeOfDay,
        breaks:Breaks[] | null,
        exceptions:Excption[]| null
}